### PR TITLE
Add Sentry Logging to Frontend Performance

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview
We want to add sentry logging for the frontend performance, so this pr introduces changes to do so at the same rate as `ifixit/fixit`. 

## CR
The `pnpm-lock.yaml` changes seemed necessary for `ts` standards, but I'm not sure how to determine issues of changing tons of packages or if I should do this a better way to reduce the code change.

## QA
You'll want to change the value of `tracesSampleRate` in `frontend/sentry.client.config.ts` to `1` to make sure 100% of logging goes to sentry. Then run `pnpm dev` and check if `react-commerce-dev` on sentry is getting performance logs.

Closes: https://github.com/iFixit/ifixit/issues/44070

### CODE REVIEW

**Review by sterlinghirsh — APPROVED** [review]

**Comment by sterlinghirsh** [comment]
CR :zap:

File: frontend/package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
       "@ifixit/shopify-storefront-client": "workspace:*",
       "@ifixit/ui": "workspace:*",
       "@sentry/nextjs": "^7.0.0",
+      "@sentry/tracing": "^7.8.1",
```
Comment (davidrans) [inline frontend/package.json:44]: Should we upgrade `@sentry/nextjs` so they're on the same version?

Comment (davidrans) [reply frontend/package.json:44]: Never mind, you did do that :+1:

**Comment by deltuh-vee** [comment]
QA 🎬 
Sentry now logs frontend performance for react-commerce.

**Pull request merged**